README.md

```diff
@@ -714,6 +714,45 @@ def test_pebble_exec():
     )
 ```
 
+### Pebble Notices
+
+Pebble can generate notices, which Juju will detect, and wake up the charm to
+let it know that something has happened in the container. The most common
+use-case is Pebble custom notices, which is a mechanism for the workload
+application to trigger a charm event.
+
+When the charm is notified, there might be a queue of existing notices, or just
+the one that has triggered the event:
+
+```python
+import ops
+import scenario
+
+class MyCharm(ops.CharmBase):
+    def __init__(self, framework):
+        super().__init__(framework)
+        framework.observe(self.on["cont"].pebble_custom_notice, self._on_notice)
+
+    def _on_notice(self, event):
+        event.notice.key  # == "example.com/c"
+        for notice in self.unit.get_container("cont").get_notices():
+            ...
+
+ctx = scenario.Context(MyCharm, meta={"name": "foo", "containers": "cont": {}})
+notices = [
+    scenario.PebbleNotice(key="example.com/a", occurences=10),
```

Comment:
@benhoyt is better placed to answer this than me, I think.

Based on the [example in the docs](https://juju.is/docs/sdk/interact-with-pebble#heading--use-custom-notices-from-the-workload-container) I think it's likely that there would be a common prefix for custom notices but it might be longer than the domain, e.g. in that example it's presumably `canonical.com/postgresql` rather than `canonical.com`. The [spec](https://docs.google.com/document/d/16PJ85fefalQd7JbWSxkRWn0Ye-Hs8S1yE99eW7pk8fA/edit) just says that it should be unique for the type of notice and that it can be 256 chars maximum.

If the `PebbleNotice` is not a custom notice but e.g. a change, then the key will just be a number if I understand correctly. So I think some repetition is likely required.

Reply:
Yeah, the keys would likely have a common prefix within a charm (but possibly not). As Tony said, that's only for custom notices, though (a `change-update` notice is coming soon), so -- if we do add this -- it would have to be `custom_notice_prefix`. I think it should be a straight string prefix and include the trailing slash, like `custom_notice_prefix='example.com/path/'` as it's more obvious.

However, I'd suggest it's simplest just to have the repetition. In most cases, I think the test would only provide a single notice anyway (the one for the event).

Reply:
ok, then let's accept the repetition for now and if need arises to add some sugar on top, we can do that later.